pallets/phala/src/stakepoolv2.rs

```diff
@@ -0,0 +1,3181 @@
+			for info in settle {
+				let payout_fixed = FixedPoint::from_bits(info.payout);
+				let reward = BalanceOf::<T>::from_fixed(&payout_fixed);
+
+				let pid = match WorkerAssignments::<T>::get(&info.pubkey) {
+					Some(pid) => pid,
+					None => {
+						Self::deposit_event(Event::<T>::RewardDismissedNotInPool {
+							worker: info.pubkey,
+							amount: reward,
+						});
+						return;
+					}
+				};
+				let mut pool_info =
+					ensure_stake_pool::<T>(pid).expect("Stake pool must exist; qed.");
+				Self::handle_pool_new_reward(&mut pool_info, reward);
+				basepool::pallet::Pools::<T>::insert(pid, PoolProxy::StakePool(pool_info));
+			}
+		}
+	}
+
+	impl<T: Config> mining::OnUnbound for Pallet<T>
+	where
+		BalanceOf<T>: FixedPointConvert + Display,
+		T: pallet_uniques::Config<CollectionId = CollectionId, ItemId = NftId>,
+		T: pallet_assets::Config<AssetId = u32, Balance = BalanceOf<T>>,
+		T: Config + vault::Config,
+	{
+		fn on_unbound(worker: &WorkerPublicKey, _force: bool) {
+			// Usually called on worker force unbinding (force == true), but it's also possible
+			// that the user unbind from the mining pallet directly.
+
+			// Warning: when using Mining & StakePool pallets together, here we assume all the
+			// miners are only registered by StakePool. So we don't bother to double check if the
+			// worker exists.
+
+			// In case of slash, `Mining::stop_mining()` will notify us a slash happened and we do
+			// bookkeeping stuff (i.e. updating releasing_stake), and eventually the slash will
+			// be enacted at `on_reclaim`.
+			Self::remove_worker_from_pool(worker);
+		}
+	}
+
+	impl<T: Config> mining::OnStopped<BalanceOf<T>> for Pallet<T>
+	where
+		BalanceOf<T>: FixedPointConvert + Display,
+		T: pallet_uniques::Config<CollectionId = CollectionId, ItemId = NftId>,
+		T: pallet_assets::Config<AssetId = u32, Balance = BalanceOf<T>>,
+		T: Config + vault::Config,
+	{
+		fn on_stopped(worker: &WorkerPublicKey, orig_stake: BalanceOf<T>, slashed: BalanceOf<T>) {}
+	}
+
+	fn pool_sub_account<T>(pid: u64, pubkey: &WorkerPublicKey) -> T
+	where
+		T: Encode + Decode,
+	{
+		let hash = crate::hashing::blake2_256(&(pid, pubkey).encode());
+		// stake pool miner
+		(b"spm/", hash)
+			.using_encoded(|b| T::decode(&mut TrailingZeroInput::new(b)))
+			.expect("Decoding zero-padded account id should always succeed; qed")
+	}
+
+	pub fn create_owner_and_lock_account<T>(pid: u64, owner: T) -> (T, T)
+	where
+		T: Encode + Decode,
+	{
+		let hash = crate::hashing::blake2_256(&(pid, owner).encode());
+		let owner_reward_account = (b"stakeowner/", hash)
+			.using_encoded(|b| T::decode(&mut TrailingZeroInput::new(b)))
+			.expect("Decoding zero-padded account id should always succeed; qed");
+		let lock_account = (b"stakelock/", hash)
+			.using_encoded(|b| T::decode(&mut TrailingZeroInput::new(b)))
+			.expect("Decoding zero-padded account id should always succeed; qed");
+		return (owner_reward_account, lock_account);
```

Comment:
```suggestion
		(owner_reward_account, lock_account)
```

Reply:
Also, please check all the account generation code to ensure it's no more than 8 bytes. Otherwise we will lose too much entropy from the hash. I suggest to make the following changes:

- `stakeowner/` -> `spo/`
- `stakelock/` -> `spl/`
- `basepool/` -> `pool/`

Reply:
done